scipy/stats/tests/test_multivariate.py

```diff
@@ -363,6 +364,86 @@ def test_frozen_dirichlet(self):
             assert_equal(d.pdf(x[:-1]), dirichlet.pdf(x[:-1], alpha))
             assert_equal(d.logpdf(x[:-1]), dirichlet.logpdf(x[:-1], alpha))
 
+    def test_numpy_rvs_shape_compatibility(self):
+        np.random.seed(2846)
+        alpha = np.array([1.0, 2.0, 3.0])
+        x = np.random.dirichlet(alpha, size=7)
+        assert_equal(x.shape, (7, 3))
+        assert_raises(ValueError, dirichlet.pdf, x, alpha)
+        assert_raises(ValueError, dirichlet.logpdf, x, alpha)
+        dirichlet.pdf(x.T, alpha)
+        dirichlet.pdf(x.T[:-1], alpha)
+        dirichlet.logpdf(x.T, alpha)
+        dirichlet.logpdf(x.T[:-1], alpha)
```

Comment:
@argriffing that issue is just changing the actual random values generated by ``rvs`` for a given seed, right? If I understood that correctly, then I'd say that technically it's still a backwards compat break but a much smaller one than returning transposed results. And if it fixes an accuracy issue for small alpha values, then I think we should consider it.